File: scripts/screenshot_dashboard.py
#!/usr/bin/env python3
"""
Dashboard Screenshot Tool
Takes a screenshot of the running dashboard for debugging
"""

import os
import time
import subprocess
from datetime import datetime
from pathlib import Path

def take_screenshot():
    """Take a screenshot of the dashboard using screencapture (macOS)"""

    # Create screenshots directory
    screenshot_dir = Path("/Users/Mike/trading/algos/EOD/tests/screenshots")
    screenshot_dir.mkdir(parents=True, exist_ok=True)

    # Generate filename
    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
    filename = screenshot_dir / f"dashboard_{timestamp}.png"
    latest_filename = screenshot_dir / "dashboard_latest.png"

    print("📷 Taking dashboard screenshot...")
    print("⏳ Please make sure the dashboard is visible on screen")
    print("   (You have 3 seconds to switch to the dashboard window)")

    # Give user time to switch to dashboard
    for i in range(3, 0, -1):
        print(f"   {i}...")
        time.sleep(1)

    # Take screenshot using macOS screencapture
    try:
        # Interactive mode - click and drag to select area
        print("📸 Click and drag to select the dashboard area...")
        subprocess.run(["screencapture", "-i", str(filename)], check=True)

        # Copy to latest
        if filename.exists():
            subprocess.run(["cp", str(filename), str(latest_filename)], check=True)
            print(f"✅ Screenshot saved: {filename}")
            print(f"✅ Latest screenshot: {latest_filename}")
            return str(latest_filename)
        else:
            print("❌ No screenshot was taken")
            return None

    except subprocess.CalledProcessError as e:
        print(f"❌ Screenshot failed: {e}")
        return None
    except Exception as e:
        print(f"❌ Unexpected error: {e}")
        return None

def main():
    """Main function"""
    screenshot_path = take_screenshot()

    if screenshot_path:
        print("\n📊 Screenshot captured successfully!")
        print(f"   View it at: {screenshot_path}")
        print("\n💡 Tip: Use Cmd+Shift+4 for quick screenshots")
        print("   or run this script again for guided capture")

if __name__ == "__main__":
    main()

File: scripts/screenshot_dashboard.js
#!/usr/bin/env node
/**
 * Dashboard Screenshot Tool
 * Takes a screenshot of the running dashboard for debugging
 */

const puppeteer = require('puppeteer');
const path = require('path');
const fs = require('fs');

async function takeScreenshot() {
    const browser = await puppeteer.launch({
        headless: true,
        defaultViewport: {
            width: 1920,
            height: 1080
        }
    });

    try {
        const page = await browser.newPage();

        // Navigate to dashboard
        console.log('📷 Navigating to dashboard...');
        await page.goto('http://127.0.0.1:8050/', {
            waitUntil: 'networkidle2',
            timeout: 30000
        });

        // Wait for chart to load
        console.log('⏳ Waiting for chart to render...');
        await page.waitForTimeout(5000);

        // Create screenshots directory if it doesn't exist
        const screenshotDir = '/Users/Mike/trading/algos/EOD/tests/screenshots';
        if (!fs.existsSync(screenshotDir)) {
            fs.mkdirSync(screenshotDir, { recursive: true });
        }

        // Take screenshot
        const timestamp = new Date().toISOString().replace(/[:.]/g, '-');
        const filename = path.join(screenshotDir, `dashboard_${timestamp}.png`);

        await page.screenshot({
            path: filename,
            fullPage: true
        });

        console.log(`✅ Screenshot saved: ${filename}`);

        // Also save a specific "latest" version for easy access
        const latestPath = path.join(screenshotDir, 'dashboard_latest.png');
        await page.screenshot({
            path: latestPath,
            fullPage: true
        });

        console.log(`✅ Latest screenshot: ${latestPath}`);

    } catch (error) {
        console.error('❌ Screenshot failed:', error);
    } finally {
        await browser.close();
    }
}

// Run the screenshot
takeScreenshot();
